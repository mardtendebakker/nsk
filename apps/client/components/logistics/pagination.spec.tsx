import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Pagination from './pagination';

function RenderC({ onNext, onPrevious } : { onNext:() => void, onPrevious:() => void }) {
  return (
    <Pagination
      date={new Date('2020-10-10')}
      onNext={onNext}
      onPrevious={onPrevious}
    />
  );
}

describe('Pagination component', () => {
  it('matches snapshot', () => {
    const { asFragment, getByText } = render(<RenderC onNext={() => {}} onPrevious={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText('10 October 2020')).toBeInTheDocument();
  });
  it('handles click properly', () => {
    const onNext = jest.fn();
    const onPrevious = jest.fn();
    const { getByTestId } = render(<RenderC onNext={onNext} onPrevious={onPrevious} />);
    fireEvent.click(getByTestId('previous'));
    fireEvent.click(getByTestId('next'));

    expect(onPrevious).toHaveBeenCalled();
    expect(onNext).toHaveBeenCalled();
  });
});
