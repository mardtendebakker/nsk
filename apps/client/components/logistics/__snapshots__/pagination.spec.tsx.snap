// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-70qvj9"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1s57ptu-MuiButtonBase-root-MuiIconButton-root"
      data-testid="previous"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ChevronLeftIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      10 October 2020
    </h4>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-qhtpox-MuiButtonBase-root-MuiIconButton-root"
      data-testid="next"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ChevronRightIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
